### PR TITLE
Denormalize clickhouse props

### PR DESCRIPTION
**Pull request by timgl**
## Changes

This gives us the option to pass through a list of properties that are denormalized as an env variable, then query those instead of via json. This is to dramatically reduce the amount of data we need to read to do queries.

Still to do
- Way to create additional denormalized columns
- use the `events_with_denormalized_props` table instead of `events` on all queries.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: ee/clickhouse/sql/events.py [hunk 1]
```diff
@@ -114,6 +123,46 @@
...
+(
+    uuid UUID,
+    event VARCHAR,
+    properties VARCHAR,
+    timestamp DateTime64(6, 'UTC'),
+    team_id Int64,
+    distinct_id VARCHAR,
+    elements_chain VARCHAR,
+    created_at DateTime64(6, 'UTC'),
+    properties_test_prop VARCHAR
```
Comment (EDsCODE) [inline ee/clickhouse/sql/events.py:137]: Is the idea here that we're manually going to rebuild the tables each time we want to put a property into a column?

Comment (timgl) [reply ee/clickhouse/sql/events.py:137]: Yep annoyingly I think that's the only way to do this, unless I'm wrong @fuziontech

Comment (fuziontech) [reply ee/clickhouse/sql/events.py]: Nope, that's what we have to do 😢

Comment (fuziontech) [reply ee/clickhouse/sql/events.py:140]: well done here btw - this is perfect

Comment (fuziontech) [reply ee/clickhouse/sql/events.py:140]: so I was doing a bit of research and found this little nugget that will make our lives _much_ easier
https://image.slidesharecdn.com/lfgouwaqrbav2qmz0z6a-signature-1b0c613dee020c35bf055baae1ebcab6567363fcb79a63f24e04d0a620019a60-poli-190816134217/95/tricks-every-clickhouse-designer-should-know-by-robert-hodges-altinity-ceo-22-638.jpg?cb=1565963144

**Review by fuziontech — APPROVED** [review]
So here's the deal with this one.

- Columns have been added to sharded_events and the distributed events table
- You can query on them asap
- Only new data will actually be materialized. History is calculated at runtime
- This means old data will be just as slow as before.
- Tested out on staging and new data is significantly faster.

**Comment by fuziontech** [comment]
I was thinking about this more and there's an easy way I can update history with these columns. Basically insert partitions into a mirror table and swap the partitions out. I'll work on that tomorrow.

**Review by fuziontech — CHANGES_REQUESTED** [review]
Blocking this because adding the columns to the distributed table caused a bug. The underlying sharded table still has it. Might just need to build a separate distributed table just for querying and keep existing one for inserting to from kafka

**Review by fuziontech — APPROVED** [review]

**Comment by fuziontech** [comment]
created a writable_events table for inserting events and events distributed table now has these columns. Should be good to go to land.

**Pull request merged**